lib/event-subscribers.loader.ts

```diff
@@ -30,27 +31,62 @@ export class EventSubscribersLoader
   loadEventListeners() {
     const providers = this.discoveryService.getProviders();
     const controllers = this.discoveryService.getControllers();
+
     [...providers, ...controllers]
-      .filter(wrapper => wrapper.isDependencyTreeStatic())
-      .filter(wrapper => wrapper.instance)
       .forEach((wrapper: InstanceWrapper) => {
-        const { instance } = wrapper;
-        const prototype = Object.getPrototypeOf(instance) || {};
-        this.metadataScanner.scanFromPrototype(
-          instance,
-          prototype,
-          (methodKey: string) =>
-            this.subscribeToEventIfListener(instance, methodKey),
-        );
+          if (wrapper.isDependencyTreeStatic() && wrapper.instance) {
```

Comment:
do mind adding a few lines of comments here or in the JSDoc for the functions bellow, explaining what we do here

Reply:
Thanks. Made it more readable and added comments